infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -477,6 +497,17 @@ private void analyzePollReportAndTriggerHandler(PollReport pollReport) {
             return;
         }
 
+        // If node has been removed. Then it should not attempt to change layout.
+        if (!latestLayout.getAllServers().contains(getLocalEndpoint()) ||
+                latestLayout.getUnresponsiveServers().contains(getLocalEndpoint())) {
+            log.warn("This Server is not a part of the active layout. Aborting failure handling.");
+            return;
+        }
+
+        // Corrects out of phase epoch issues if present in the report. This method performs
+        // re-sealing of all nodes if required and catchup of a layout server to the current state.
+        correctOutOfPhaseEpochs(pollReport);
+
         final ManagementClient localManagementClient = corfuRuntime.getRouter(getLocalEndpoint())
                 .getClient(ManagementClient.class);
 
```

Comment:
The heal action should be separate handler.

Reply:
Done